bsp/src/org/jetbrains/bsp/project/resolver/BspResolverLogic.scala

```diff
@@ -220,8 +213,7 @@ private[resolver] object BspResolverLogic {
     val sourceRoots = filterRoots(sourceDirs)
     val resourceRoots = filterRoots(resourceDirs)
 
-    val moduleBase = Option(target.getBaseDirectory)
-      .map(_.toURI.toFile)
+    val moduleBase = Option(target.getId.getUri.toURI.toFile)
```

Comment:
Actually this fails when the `id` URI is not a file, which is valid in BSP. Why did you change it to this?